clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy

```diff
@@ -26,6 +26,8 @@ class AzureAppGatewayDescription extends AzureResourceOpsDescription {
   String loadBalancerName
   String vnet
   String subnet
+  Boolean hasNewSubnet // used for tracking the subnets that are automatically created and are not selected by the user
+  Boolean useDefaultVnet = false // default is for the user to provide us with an existing vnet and subnet
```

Comment:
As of now there's nothing coming from deck that will automatically create a vnet/subnet for you; implementation to have such an option is a nice thing to have but outside of current requirements. The only viable option is to directly call into clouddriver via "curl" in order to overwrite this settings; this could be useful when trying to automate tests that create/delete resources. It could also be use when creating and provisioning an Azure Spinnaker VM that has some pipelines which can be executed (for demo purposes).